runtime/tests/ecdsa.rs

```
@@ -12,140 +11,138 @@ use zerocopy::AsBytes;
+                    id,
+                    comment: comment.to_owned(),
+                });
+            }
 
-        cmd.chksum = caliptra_common::checksum::calc_checksum(
-            u32::from(CommandId::ECDSA384_VERIFY),
-            &cmd.as_bytes()[4..],
-        );
+            report_test(&mut wyche_ran, test.tc_id, &test.comment);
+            model
+                .compute_sha512_acc_digest(test.msg.as_slice(), ShaAccMode::Sha384Stream)
+                .unwrap();
 
-        let resp = model.mailbox_execute(u32::from(CommandId::ECDSA384_VERIFY), cmd.as_bytes());
-        assert_eq!(
-            resp,
-            Err(ModelError::MailboxCmdFailed(
-                CaliptraError::RUNTIME_ECDSA_VERIFY_FAILED.into()
-            ))
-        );
+            let cmd = EcdsaVerifyCmd {
+                chksum: 0,
+                pub_key_x: test_group.key.affine_x.as_slice()[..].try_into().unwrap(),
+                pub_key_y: test_group.key.affine_y.as_slice()[..].try_into().unwrap(),
+                signature_r: signature
+                    .r()
+                    .to_vec_padded(48)
+                    .unwrap()
+                    .as_bytes()
+                    .try_into()
+                    .unwrap(),
+                signature_s: signature
+                    .s()
+                    .to_vec_padded(48)
+                    .unwrap()
+                    .as_bytes()
+                    .try_into()
+                    .unwrap(),
+                // Do tests on mailbox
+            };
+            let checksum = caliptra_common::checksum::calc_checksum(
+                u32::from(CommandId::ECDSA384_VERIFY),
+                &cmd.as_bytes()[4..],
+            );
+            let cmd = EcdsaVerifyCmd {
+                chksum: checksum,
+                ..cmd
+            };
+            match model.mailbox_execute(u32::from(CommandId::ECDSA384_VERIFY), cmd.as_bytes()) {
+                Ok(resp) => match test.result {
```

Instead of checking `if resp.is_some()`, you can match `Ok(Some(resp))`. if `Ok(None)` matches, we should probably panic since that is unexpected.